com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PollExecuteAppStatusExecution.java

```diff
@@ -37,39 +29,32 @@
     private static final String DEFAULT_SUCCESS_MARKER = "STDOUT:SUCCESS";
     private static final String DEFAULT_FAILURE_MARKER = "STDERR:FAILURE";
 
-    @Inject
     protected RecentLogsRetriever recentLogsRetriever;
 
-    @Override
-    public String getLogicalStepName() {
-        return StartAppStep.class.getSimpleName();
-    }
-
-    @Override
-    public String getIndexVariable() {
-        return Constants.VAR_APPS_INDEX;
+    public PollExecuteAppStatusExecution(RecentLogsRetriever recentLogsRetriever) {
+        this.recentLogsRetriever = recentLogsRetriever;
     }
 
     @Override
-    protected ExecutionStatus executeStepInternal(DelegateExecution context) throws Exception {
-        getStepLogger().logActivitiTask();
-        CloudApplication app = getNextApp(context);
+    public AsyncExecutionState execute(ExecutionWrapper execution) throws Exception {
+        execution.getStepLogger().logActivitiTask();
+        CloudApplication app = getNextApp(execution.getContext());
         ApplicationAttributesGetter attributesGetter = ApplicationAttributesGetter.forApplication(app);
         boolean executeApp = attributesGetter.getAttribute(SupportedParameters.EXECUTE_APP, Boolean.class, false);
 
         if (!executeApp) {
-            return ExecutionStatus.SUCCESS;
+            return AsyncExecutionState.FINISHED;
         }
         try {
-            CloudFoundryOperations client = getCloudFoundryClient(context);
-            Pair<AppExecutionStatus, String> status = getAppExecutionStatus(context, client, attributesGetter, app);
-            return checkAppExecutionStatus(context, client, attributesGetter, app, status);
+            CloudFoundryOperations client = execution.getCloudFoundryClient();
+            Pair<AppExecutionStatus, String> status = getAppExecutionStatus(execution.getContext(), client, attributesGetter, app);
+            return checkAppExecutionStatus(execution, client, attributesGetter, app, status);
         } catch (CloudFoundryException cfe) {
             SLException e = StepsUtil.createException(cfe);
-            getStepLogger().error(e, Messages.ERROR_EXECUTING_APP_1, app.getName());
-            throw e;
+            execution.getStepLogger().error(e, Messages.ERROR_EXECUTING_APP_1, app.getName());
+            throw cfe;
```

Comment:
Why throw the CloudFoundryException and not the SLException? This seems to repeat in a lot of places, so I'm not going to comment on each and every one of them.